File: 1147.段式回文/1147-段式回文.py
class Solution:
    def longestDecomposition(self, s: str) -> int:
        if s == "":
            return 0
        for i in range(1, len(s) // 2 + 1):  # 枚举前后缀长度
            if s[:i] == s[-i:]:  # 立刻分割
                return 2 + self.longestDecomposition(s[i:-i])
        return 1  # 无法分割

File: 1147.段式回文/1147-段式回文.js
/**
 * @param {string} text
 * @return {number}
 */
var longestDecomposition = function(text) {
    if(text==='') return 0;
    if(text.length===1) return 1; 
    const n=text.length;
    // console.log(text, n);
    for(let i=1; i<=Math.floor((n+1)/2); i++){
        if(text.slice(0, i)===text.slice(n-i, n)){
            return 2+longestDecomposition(text.slice(i, n-i));
        }
    }
    return 1;
};